### PR TITLE
Add markdownlint to pre-commit

### PR DESCRIPTION
**Pull request by irisliucy**
Add markdownlint to pre-commit hook. This will be helpful especially when writing documentation/ README using markdown file.

### CODE REVIEW

**Comment by irisliucy** [comment]
@Mergifyio rebase

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -54,6 +54,11 @@ repos:
     -   id: pretty-format-json               # Checks that all your JSON files have keys that are sorted and indented.
     -   id: trailing-whitespace              # Trims trailing whitespace.
 
+-   repo: https://github.com/jumanjihouse/pre-commit-hooks
```
Comment (ryanjulian) [inline .pre-commit-config.yaml:57]: all else equal, i think we should use the officially-provided hook
https://github.com/markdownlint/markdownlint/blob/master/.pre-commit-hooks.yaml

is there a reason you picked this one?

Comment (irisliucy) [reply .pre-commit-config.yaml:57]: Unfortunately, no. I came across a list of pre-commit hooka here: https://pre-commit.com/hooks.html

I will switch it to the official one.

**Comment by irisliucy** [comment]
@gitanshu Sure, markdown config is added to `.editorconfig`. Also, I use atom prior to switching to vscode so I totally get it.

**Comment by irisliucy** [comment]
@Mergifyio rebase

**Comment by irisliucy** [comment]
@Mergifyio rebase

**Comment by ryanjulian** [comment]
@Mergifyio rebase

**Comment by irisliucy** [comment]
Added a `mdlrc` markdown config file . I disable some uncommon setting, such as line length for `.md` = 80, unordered list, multiple consecutive blank lines etc. 

Happy doc writing in markdown!

**Comment by irisliucy** [comment]
@Mergifyio rebase

File: docker/Dockerfile [hunk 2]
```diff
@@ -33,6 +33,8 @@ RUN \
     libreadline-dev \
     libssl-dev \
     libsqlite3-dev && \
```
Comment (gitanshu) [inline docker/Dockerfile:35]:
```suggestion
    libsqlite3-dev \
```

File: docker/Dockerfile [hunk 3]
```diff
@@ -33,6 +33,8 @@ RUN \
...
+    # ruby
+    ruby-full \
```
Comment (gitanshu) [inline docker/Dockerfile:37]:
```suggestion
    ruby-full && \
```

**Pull request merged**